### PR TITLE
Set policyAsset for core unit tests

### PR DESCRIPTION
**Pull request by nkostoulas**
Set policyAsset for all core c++ unit tests #408 

src/test/txvalidationcache_tests.cpp was failing with memory access violation somewhere in block creation. I believe it's because now that mempool entry fee is correctly set, the fact that the fee output was larger than the non-fee one was causing an issue. I haven't been able to track this down exactly though. Please let me know if you have some idea about this and if the change makes sense.

### CODE REVIEW

**Comment by jtimon** [comment]
utACK 2bf55e1d0a91fdb34e1af8e84a67e5e01fb62818

File: src/test/txvalidationcache_tests.cpp [hunk 1]
```diff
@@ -45,10 +45,10 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, TestChain100Setup)
         spends[i].vin[0].prevout.hash = coinbaseTxns[0].GetHash();
         spends[i].vin[0].prevout.n = 0;
         spends[i].vout.resize(2);
-        spends[i].vout[0].nValue = 11*CENT;
+        spends[i].vout[0].nValue = MAX_MONEY-11*CENT;
```
Comment (instagibbs) [inline src/test/txvalidationcache_tests.cpp]: why this change?

Comment (instagibbs) [reply src/test/txvalidationcache_tests.cpp]: nevermind
